File: OnlineStoreAdminPanel/OnlineStoreAdminPanel.Web/Modules/ServerTypes/OnlineStore/CatigoryColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CatigoryRow } from "./CatigoryRow";

export interface CatigoryColumns {
    Id: Column<CatigoryRow>;
    Name: Column<CatigoryRow>;
    Photo: Column<CatigoryRow>;
    Description: Column<CatigoryRow>;
}

export class CatigoryColumns extends ColumnsBase<CatigoryRow> {
    static readonly columnsKey = 'OnlineStore.Catigory';
    static readonly Fields = fieldsProxy<CatigoryColumns>();
}

File: OnlineStoreAdminPanel/OnlineStoreAdminPanel.Web/Modules/OnlineStore/Catigory/CatigoryColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace OnlineStoreAdminPanel.OnlineStore.Columns;

[ColumnsScript("OnlineStore.Catigory")]
[BasedOnRow(typeof(CatigoryRow), CheckNames = true)]
public class CatigoryColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Photo { get; set; }
    public string Description { get; set; }
}